src/serializer.ts

```diff
@@ -131,7 +128,7 @@ export class Serializer {
             }
         }
         if (children[discriminatorValue] === undefined) {
-            return parent;
+            throw new TypeError(`No class for ${parent.name} class with discriminator value ${obj[discriminatorField]}`);
```

Comment:
I would add some "child" or "sub" notion in the message. Something like `"No matching subclass for parent class ${parent.name} with discriminator value ${obj[discriminatorField]}"`